gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/extractor/FileAwareInputStreamExtractorWithCheckSchema.java

```diff
@@ -39,32 +43,38 @@
  * check if the schema matches the expected schema. If not it will abort the job.
  */
 
-public class FileAwareInputStreamExtractorWithCheckSchema extends FileAwareInputStreamExtractor{
+public class FileAwareInputStreamExtractorWithCheckSchema extends FileAwareInputStreamExtractor {
```

Comment:
just a comment on this class: i understand that this class already exists and its adding proper field level schema check, but can schemaCheck not be part of `FileAwareInputStreamExtractor` itself instead of introducing another class for schemaCheck functionality, since that is the only difference with this class which probably can be controlled by a schemaCheck flag. may be its a separate PR.

Reply:
@jhsenjaliya the reason we did this in a different class is because it is leaking Avro details. CopySource and the associated extractor are able to copy arbitrary files simply as byte streams. This particular extension requires the copied files to be Avro.

In general, I think that in a byte copier any functionality that assumes a schema associated with those bytes is pollution of the logic. What do you think?

Reply:
ok, sure. Thanks